src/app/shared/comment-editor/comment-editor.component.ts

```diff
@@ -69,6 +68,56 @@ export class CommentEditorComponent implements OnInit {
     this.commentField.setValidators([Validators.maxLength(this.maxLength)]);
   }
 
+  onKeyPress(event) {
+    if (event.ctrlKey) {
+      const selectionStart = this.commentTextArea.nativeElement.selectionStart;
+      const selectionEnd = this.commentTextArea.nativeElement.selectionEnd;
+      let insertedChar = '';
+      switch (event.key) {
+        case 'b':
+          event.preventDefault();
+          insertedChar = '**';
+          break;
+        case 'i':
+          event.preventDefault();
+          insertedChar = '*';
+          break;
+        default:
+          return;
+      }
+      const currentText = this.commentField.value;
+      const highlightedText = currentText.slice(selectionStart, selectionEnd);
+      const hasInsertedCharBefore = currentText.slice(selectionStart - insertedChar.length, selectionStart) === insertedChar;
+      const hasInsertedCharAfter = currentText.slice(selectionEnd, selectionEnd + insertedChar.length) === insertedChar;
+      if (hasInsertedCharBefore && hasInsertedCharAfter) {
```

Comment:
There is a toggling being done here, this can be indicated in the private method name or in a comment. (Toggle bold/italics)

Reply:
Good point @gycgabriel ,to add on I think another suggestion is to extract out this condition into another method/boolean variable e.g. 'alreadyFormatted' (Please go ahead and use another name if you prefer, I can't think of a good one 😅)

Reply:
Good suggestions @gycgabriel and @kkangs0226, i've abstracted this out into a private method